pkgs/initiator/initiator.go

```diff
@@ -280,8 +293,103 @@ func (c *Initiator) StartDKG(id [24]byte, withdraw []byte, ids []uint64, network
+	// make resign message
+	rMsg := &wire.ResignMessage{
+		Operators: ops,
+		Resign: &spec.Resign{ValidatorPubKey: proofs[0].Proof.ValidatorPubKey,
+			Fork:                  network.GenesisForkVersion(),
+			WithdrawalCredentials: withdraw,
+			Owner:                 owner,
+			Nonce:                 nonce},
+		Proofs: proofs,
+	}
+	c.Logger.Info("Outgoing resign request fields",
+		zap.String("network", hex.EncodeToString(rMsg.Resign.Fork[:])),
+		zap.String("withdrawal", hex.EncodeToString(rMsg.Resign.WithdrawalCredentials)),
+		zap.String("owner", hex.EncodeToString(rMsg.Resign.Owner[:])),
+		zap.Uint64("nonce", rMsg.Resign.Nonce),
+		zap.Any("operator IDs", ids))
+	for _, proof := range rMsg.Proofs {
+		c.Logger.Info("Loaded proof",
+			zap.String("ValidatorPubKey", hex.EncodeToString(proof.Proof.ValidatorPubKey)),
+			zap.String("Owner", hex.EncodeToString(proof.Proof.Owner[:])),
+			zap.String("SharePubKey", hex.EncodeToString(proof.Proof.SharePubKey)),
+			zap.String("EncryptedShare", hex.EncodeToString(proof.Proof.EncryptedShare)),
+			zap.String("Signature", hex.EncodeToString(proof.Signature)))
+	}
+	c.Logger = c.Logger.With(instanceIDField)
+	resultsBytes, err := c.resignMessageFlowHandling(rMsg, id, ops)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	dkgResults, err := parseDKGResultsFromBytes(resultsBytes, id)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	depositDataJson, keyshares, err := c.processDKGResultResponse(dkgResults, id, rMsg.Operators, rMsg.Resign.WithdrawalCredentials, rMsg.Resign.Fork, rMsg.Resign.Owner, rMsg.Resign.Nonce)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	c.Logger.Info("✅ verified master signature for ssv contract data")
+	if err := crypto.ValidateDepositDataCLI(depositDataJson, common.BytesToAddress(withdraw)); err != nil {
+		return nil, nil, nil, err
+	}
+	if err := crypto.ValidateKeysharesCLI(keyshares, rMsg.Operators, rMsg.Resign.Owner, rMsg.Resign.Nonce, depositDataJson.PubKey); err != nil {
+		return nil, nil, nil, err
+	}
+	// sending back to operators results
+	depositData, err := json.Marshal(depositDataJson)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	keysharesData, err := json.Marshal(keyshares)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	var proofsArray []*wire.SignedProof
+	for _, res := range dkgResults {
+		proofsArray = append(proofsArray, &wire.SignedProof{res.SignedProof}) //nolint:all
+	}
+	proofsData, err := json.Marshal(proofsArray)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	resultMsg := &wire.ResultData{
+		Operators:     ops,
+		Identifier:    id,
+		DepositData:   depositData,
+		KeysharesData: keysharesData,
+		Proofs:        proofsData,
+	}
+	err = c.sendResult(resultMsg, ops, consts.API_RESULTS_URL, id)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("🤖 Error storing results at operators %w", err)
+	}
+	return depositDataJson, keyshares, proofsArray, nil
```

Comment:
The codes here after operators resigning are the same with **StartDKG** above (line 254-293), for better coding practice and future maintainance, maybe wrap it in another function like "processDKGResults"?